src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+            tail = head;
+            size++;
+        } else {
+            node = new Node<>(tail, value, null);
+            node.prev.next = node;
+            tail = node;
+            size++;
+        }
```

Comment:
When creating a new `Node<T>`, you should pass all the fields to the `Node` constructor. The `node` variable is being reassigned on line 23 which is unnecessary. You can directly use the tail reference and update its next node.